src/NUnitFramework/testdata/TestFixtureSourceData.cs

```diff
@@ -353,6 +354,69 @@ public void SomeTest()
         }
     }
 
+    public class GenericFixtureWithTypeAndConstructorArgsSource
+    {
+        public static readonly ITestFixtureData[] Source =
+        {
+            new TypedTestFixture<int>(5),
+            new TypedTestFixture<object>(new object())
+        };
+
+        public class TypedTestFixture<T> : Framework.Internal.TestParameters, ITestFixtureData
+        {
+            public TypedTestFixture(params object[] arguments)
+                : base(arguments)
+            {
+                TypeArgs = new[] { typeof(T) };
+            }
+
+            public Type[] TypeArgs { get; }
+        }
+    }
+
+    [TestFixtureSource(typeof(GenericFixtureWithTypeAndConstructorArgsSource), "Source")]
```

Comment:
What about `"Source"` => `nameof(GenericFixtureWithTypeAndConstructorArgsSource.Source)`?

Reply:
I implemented it like that at first but then I wasn't sure whether that would be supported by the compiler c# level.
All other examples in the same file use constant strings instead of the  `nameof(...)` operator.
If `nameof` is fine I'll create a separate pull request to modify all string references to  types/members to use `nameof(...)` - that would make sense, right?

Reply:
@jnm2 thanks!

-----------

I'd be happy to provide a PR to change string-constants to `nameof(..)`, only adjust the newly introduced usages, or open a new issue to discuss the topic.
However, I'm not in a position to decide which is the right way to go so I'll refrain from doing anything regarding this topic until I get instructions ;-)